### PR TITLE
feat: upgrade Spring to v6, SpringBoot to v3

### LINKED ISSUE #649: influxdb-spring 6.10.0 does not work with spring boot 3.2.0
it throws exception:

```
Caused by: java.lang.NoSuchMethodError: org.springframework.boot.actuate.autoconfigure.health.CompositeHealthContributorConfiguration: method 'void <init>()' not found
	at com.influxdb.spring.health.InfluxDB2HealthIndicatorAutoConfiguration.<init>(InfluxDB2HealthIndicatorAutoConfiguration.java:50) ~[influxdb-spring-6.10.0.jar:6.10.0]
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:197) ~[spring-beans-6.1.1.jar:6.1.1]
```

### PR DESCRIPTION
**Pull request by bednar**
Closes #649

## Proposed Changes

Migrate `influxdb-spring` from Spring Boot v2 to v3, Spring to v6.

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] `mvn test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -393,7 +393,7 @@ public class FluxExample {
 
 ## Build Requirements
 
-* Java 1.8+ (tested with jdk8)
+* Java 17+ (tested with JDK 17)
```
Comment (powersj) [inline README.md:396]: I know this is a draft, but I wanted to put this down so I don't forget. Can we add a note for users who are stuck on JDK 8 and JDK 11, to use the previous version?

Comment (bednar) [reply README.md:396]: Of course, I will fix the tests and then update the CHANGELOG and README documentation.

Comment (bednar) [reply README.md:396]: done

File: client/src/test/java/com/influxdb/client/WriteApiTest.java [hunk 2]
```diff
@@ -1012,7 +1013,15 @@ void userAgent() throws InterruptedException {
 
         String userAgent = recordedRequest.getHeader("User-Agent");
 
-        Assertions.assertThat(userAgent).startsWith("influxdb-client-java/7.");
+        String currentVersion = UserAgentInterceptor.class.getPackage().getImplementationVersion();
```
Comment (karel-rehor) [inline client/src/test/java/com/influxdb/client/WriteApiTest.java:1016]: Added a test change here.  Jakub, does this work for you?  Does it not defeat Utku Aydin's intent for this assertion?

**Review by karel-rehor — APPROVED** [review]
This looks OK.  To check the build of this branch locally I added a change to WriteApiTest.userAgent():1004-1025.  If this change does not negatively impact earlier intentions for this test, I think this PR can be merged.

**Pull request merged**